source/PlayState.hx

```diff
@@ -1,21 +1,72 @@
 package;
 
 import actors.player.Hero;
+import environment.Door;
+import environment.TotalKeys;
+import environment.Key;
+import flixel.group.FlxGroup.FlxTypedGroup;
 import flixel.FlxState;
+import flixel.FlxG;
+import flixel.FlxObject;
 
 class PlayState extends FlxState
 {
 	private var hero:Hero;
+	private var doors:FlxTypedGroup<Door>;
+	private var door:Door;
+	private var keys:FlxTypedGroup<Key>;
+	private var key:Key;
+	private var totalKeys:TotalKeys;
 
 	override public function create():Void
 	{
 		super.create();
 		hero = new Hero();
 		add(hero);
+
+		// Add door objects
+		doors = new FlxTypedGroup<Door>();
+		door = new Door(50, 50);
+		doors.add(door);
+		door = new Door(250, 250);
+		doors.add(door);
+		add(doors);
+
+		// Add key objects
+		keys = new FlxTypedGroup<Key>();
+		key = new Key(100, 100);
+		keys.add(key);
+		key = new Key(300, 300);
+		keys.add(key);
+		add(keys);
+
+		totalKeys = new TotalKeys();
 	}
 
 	override public function update(elapsed:Float):Void
 	{
+		// Checks if hero is collideing with doors or keys
+		FlxG.overlap(hero, doors, openDoor);
+		FlxG.overlap(hero, keys, pickupKey);
+		FlxG.collide(hero, doors);
+		
 		super.update(elapsed);
```

Comment:
The way that the Hero object kinda jitters when walking into locked doors is caused by the order of the statements here. 

Remember that, because the PlayState is ultimately a child of FlxGroup, `super.update()` really is just saying "update all of the things that have been added to me"

The current order of statements results in the following:
 * Check if hero and door are colliding, if so, separate them!
 * Now update things inside me (including Hero)
 * Hero's update function notices a directional key is held, moves in that direction without realizing it's overlapping a door.

What we actually want is for the Hero's update function to happen first, then do all of the overlap and collision logic. When we order things this way, the player doesn't see the inbetween part where the Hero is overlapping with the locked door, that all gets cleaned up before the player ends up seeing it.